server/src/main/java/io/spine/server/storage/QueryConverter.java

```diff
@@ -0,0 +1,309 @@
+
+        identifiers(builder, filters.getIdFilter());
+        filters(builder, spec, filters);
+        fieldMask(builder, format);
+        orderByAndLimit(builder, spec, format);
+
+        return builder.build();
+    }
+
+    /**
+     * Creates a new {@code RecordQuery} based on the given response format and
+     * the storage specification of the queried record.
+     *
+     * <p>The result contains no filters on any record field.
+     *
+     * @param spec
+     *         the specification of the columns stored for the particular record type
+     * @param format
+     *         the original response format
+     * @param <I>
+     *         the type of the record identifiers
+     * @param <R>
+     *         the type of the records which are queried
+     * @return a new record query
+     */
+    public static <I, R extends Message> RecordQuery<I, R>
+    newQuery(RecordSpec<I, R, ?> spec, ResponseFormat format) {
+        checkNotNull(spec);
+        checkNotNull(format);
+
+        Class<I> idType = spec.idType();
+        Class<R> recordType = spec.recordType();
+        RecordQueryBuilder<I, R> builder = RecordQuery.newBuilder(idType, recordType);
+        fieldMask(builder, format);
+        orderByAndLimit(builder, spec, format);
+        return builder.build();
+    }
+
+    private static <I, R extends Message> void
+    filters(RecordQueryBuilder<I, R> builder, RecordSpec<I, R, ?> spec, TargetFilters filters) {
+        if (filters.getFilterCount() > 0) {
+            List<CompositeFilter> compositeFilters = filters.getFilterList();
+            for (CompositeFilter composite : compositeFilters) {
+                CompositeFilter.CompositeOperator compositeOperator = composite.getOperator();
+
+                switch (compositeOperator) {
+                    case ALL:
+                        convert(composite, spec, builder);
+                        break;
+                    case EITHER:
+                        builder.either((either) -> convert(composite, spec, either));
+                        break;
+                    case UNRECOGNIZED:
+                    case CCF_CO_UNDEFINED:
+                    default:
+                        throw newIllegalArgumentException(
+                                "Unsupported composite operator `%s` encountered.",
+                                compositeOperator);
+                }
+            }
+        }
+    }
+
+    /**
+     * Converts the passed composite filter into a set of conditions and adds them to the passed
+     * builder according to the record specification.
+     */
+    @CanIgnoreReturnValue
+    private static <I, R extends Message> RecordQueryBuilder<I, R>
+    convert(CompositeFilter filter, RecordSpec<I, R, ?> spec, RecordQueryBuilder<I, R> builder) {
+        if (filter.getFilterCount() > 0) {
+            List<Filter> childFilters = filter.getFilterList();
+            for (Filter childFilter : childFilters) {
```

Comment:
Again, please consider inlining the `childFilter` and removing `if` check to make this code less nested.